### PR TITLE
Fixed scheduling and cancelling receive timeout #25340

### LINKED ISSUE #25340: ReceiveTimeout is still not triggered in this case
I have a very similar to #24770 issue.

In my case `setReceiveTimeout(...)` is called by code which handles a message marked as `NotInfluenceReceiveTimeout`. The `ReceiveTimeout` message will never be sent to the actor.

```scala
import akka.actor.{Actor, ActorSystem, NotInfluenceReceiveTimeout, Props, ReceiveTimeout}
import scala.concurrent.duration._

object ReceiveTimeoutTestApp extends App {
  val sys = ActorSystem()
  val a = sys.actorOf(Props[MyActor])

  a ! Message
}

class MyActor extends Actor {

  override def receive: Receive = {
    case Message ⇒
      println("Message")
      context.setReceiveTimeout(1.second)

    case ReceiveTimeout ⇒
      println("ReceiveTimeout")
      self ! Message
  }
}

// Breaks when Message extends NotInfluenceReceiveTimeout
object Message //extends NotInfluenceReceiveTimeout
```

### PR DESCRIPTION
**Pull request by nyukhalov**
Resolves #25340

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @nyukhalov,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by johanandren** [comment]
OK TO TEST

**Comment by nyukhalov** [comment]
Hey guys.
It is my first contribution to akka, so I probably missed something.
Can you help me to check why did `Jenkins PR Validation` failed if there are no failed tests?

Also, `continuous-integration/travis-ci/pr` failed with `Binary compatibility check failed!` as I touched a signature of the `checkReceiveTimeout` method in the `ReceiveTimeout` trait. I thought it is not an issue as it is an internal trait `private[akka] trait ReceiveTimeout`

**Comment by johanandren** [comment]
Hi @nyukhalov, thanks for contributing! Check out the contribution guidelines for more details about the bincomp check (it doesn't understand what is internal): https://github.com/akka/akka/blob/master/CONTRIBUTING.md

As you say, if it is internal you can add a filter entry for it. You can also check locally that you got it right by running the task `mimaReportBinaryIssues` in sbt.

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
Sorry for the reviewing taking time, vacation times so we are thin in the team this and the coming weeks.

File: akka-actor/src/main/mima-filters/2.5.13.backwards.excludes [hunk 1]
```diff
@@ -14,3 +14,9 @@ ProblemFilters.exclude[IncompatibleResultTypeProblem]("akka.io.SimpleDnsCache#Ex
 ProblemFilters.exclude[IncompatibleMethTypeProblem]("akka.io.SimpleDnsCache#ExpiryEntry.this")
 ProblemFilters.exclude[MissingClassProblem]("akka.io.SimpleDnsCache$ExpiryEntryOrdering$")
 ProblemFilters.exclude[IncompatibleMethTypeProblem]("akka.io.SimpleDnsCache#Cache.get")
+
+#25347 - Fixed scheduling and cancelling receive timeout
+ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.ActorCell.checkReceiveTimeout")
+ProblemFilters.exclude[DirectMissingMethodProblem]("akka.actor.dungeon.ReceiveTimeout.checkReceiveTimeout")
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.dungeon.ReceiveTimeout.checkReceiveTimeout$default$1")
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.actor.dungeon.ReceiveTimeout.checkReceiveTimeout")
```
Comment (chbatey) [inline akka-actor/src/main/mima-filters/2.5.13.backwards.excludes]: These files should have a new line at the end

**Review by chbatey — APPROVED** [review]
LGTM once line is added to the excludes

Thanks!

**Comment by johanandren** [comment]
I added the linefeed so we can merge after validation, no need to do anything more @nyukhalov 
Thanks for the PR!

**Comment by nyukhalov** [comment]
@johanandren @chbatey thank you

**Pull request merged**